Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -37,34 +36,55 @@ bool LogFileReader::readAll(const QString& fileName)
         if (!parseLine(line, batteryData))
         {
             qDebug() << "Error while parsing" << line;
-            // return false;
+             return false;
         }
         else
         {
-            // This is how to send out a signal in QT using the emit keyword.
-            // This line notifies the classes listening to this signal
-            // that battery data has been received.
             emit batteryDataReceived(batteryData);
         }
     }
 
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
+
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if (sections.length() < COLUMNS)
+    {
+        qDebug() << "sections length is less than 3";
```

Comment:
We don't really need these Debug messages, since readAll() already prints out one when `parseLine` returns false.
Also because readAll()'s message will print out after, so reading `batDataFailure.csv` would look like this:
`Voltage: 3.6 Current: -12.1075 Total Ah used: 61 Time remaining until the Battery is Charged: 18:57:42.494`
`time is invalid`
`Error while parsing "07:00,04.926, 3.596829, -2.408174"`
Which looks a bit funky, since it looks like it's saying that the first line is invalid, when it's really the next one.